includes/class-wc-stripe-webhook-handler.php

```diff
@@ -292,13 +292,16 @@ public function process_webhook_dispute( $notification ) {
 			return;
 		}
 
+		// Sometimes (at least in testing) the "dispute.created" and "charge.succeeded" events can arrive really close for SEPA payments.
+		// Use the locking system to prevent the charge.succeeded to set the order to "processing" even after the dispute was created.
+		$this->lock_order_payment( $order );
+
 		/* translators: 1) The URL to the order. */
-		$order->update_status( 'on-hold', sprintf( __( 'A dispute was created for this order. Response is needed. Please go to your <a href="%s" title="Stripe Dashboard" target="_blank">Stripe Dashboard</a> to review this dispute.', 'woocommerce-gateway-stripe' ), $this->get_transaction_url( $order ) ) );
+		$order->update_status( 'failed', sprintf( __( 'A dispute was created for this order. Response is needed. Please go to your <a href="%s" title="Stripe Dashboard" target="_blank">Stripe Dashboard</a> to review this dispute.', 'woocommerce-gateway-stripe' ), $this->get_transaction_url( $order ) ) );
 
 		do_action( 'wc_gateway_stripe_process_webhook_payment_error', $order, $notification );
 
-		$order_id = WC_Stripe_Helper::is_wc_lt( '3.0' ) ? $order->id : $order->get_id();
-		$this->send_failed_order_email( $order_id );
```

Comment:
As I see those lines would send an email about failed order. Why do we change that?

Reply:
When the order is set to `failed`, the "Order Failed" email is already triggered automatically, so these lines are redundant (in fact, they would send another email).

Reply:
If that's true, then extra emails are already being sent in certain other cases, e.g.:

https://github.com/woocommerce/woocommerce-gateway-stripe/blob/19e5f4a47d4ad94d4d367a588b3cd4b227c9f80d/includes/class-wc-stripe-webhook-handler.php#L673-L678

Should those `send_failed_order_email` calls be removed? Can open an issue to look into that.

Reply:
> Should those send_failed_order_email calls be removed?

I think they should, yes.

Reply:
👍 Opened https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1229